pkg/container-collection/options.go

```diff
@@ -796,26 +793,32 @@ func WithTracerCollection(tc TracerCollection) ContainerCollectionOption {
 		}()
 
 		cc.containerEnrichers = append(cc.containerEnrichers, func(container *Container) bool {
-			path := filepath.Join(host.HostProcFs, fmt.Sprint(container.Pid), "ns", "mnt")
-			fd, err := unix.Open(path, unix.O_RDONLY|unix.O_CLOEXEC, 0)
+			mntNsPath := filepath.Join(host.HostProcFs, fmt.Sprint(container.Pid), "ns", "mnt")
+			mntNsFd, err := unix.Open(mntNsPath, unix.O_RDONLY|unix.O_CLOEXEC, 0)
 			if err != nil {
 				log.Warnf("WithTracerCollection: failed to open mntns reference for container %s: %s",
 					container.ID, err)
 				return false
 			}
+			container.mntNsFd = mntNsFd
 
-			container.mntNsFd = fd
+			netNsPath := filepath.Join(host.HostProcFs, fmt.Sprint(container.Pid), "ns", "net")
+			netNsFd, err := unix.Open(netNsPath, unix.O_RDONLY|unix.O_CLOEXEC, 0)
+			if err != nil {
+				log.Warnf("WithTracerCollection: failed to open netns reference for container %s: %s",
+					container.ID, err)
+				unix.Close(mntNsFd)
+				return false
```

Comment:
What about removing unix.Close() from here and add the new container.close() function in two places:
* https://github.com/inspektor-gadget/inspektor-gadget/blob/2a2630c73bac7c7cd51727785dc61167d9bb4406/pkg/container-collection/container-collection.go#L108-L110
* https://github.com/inspektor-gadget/inspektor-gadget/blob/2a2630c73bac7c7cd51727785dc61167d9bb4406/pkg/container-collection/container-collection.go#L168-L170

```
// Enrichers can decide to drop a container
```